### PR TITLE
MP/ Refactor product detail internals and move it to Store module

### PR DESCRIPTION
**Pull request by MPucheta**

### CODE REVIEW

**Review by eigiachero — APPROVED** [review]

**Review by roke77 — APPROVED** [review]

File: src/modules/MainApp/components/Arrows/Arrow.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+const getDirection = (color, direction) => {
+  if (color === 'yellow') {
+    if (direction === 'left') return ArrowLeftBlack
+
+    return ArrowRightBlack
+  }
+
+  if (direction === 'left') return ArrowLeftYellow
+
+  return ArrowRightYellow
```
Comment (nahudalla) [inline src/modules/MainApp/components/Arrows/Arrow.js]: This is returning `Black` arrow when `color === 'yellow'` and vice versa, is this intended?

File: src/modules/MainApp/components/Slider/index.js [hunk 2]
```diff
@@ -1,24 +1,22 @@
 import React from 'react'
 
 import DotSelected from './DotSelected'
-import DotUnSelected from './DotUnSelected'
+import DotUnselected from './DotUnselected'
```
Comment (nahudalla) [inline src/modules/MainApp/components/Slider/index.js]: Are you sure we want two components that do exactly the same, look exactly the same, but only change a minor style? Why not have a `Dot` component and receive a `selected` boolean prop to change the styling?

File: src/modules/MainApp/components/Slider/DotSelected.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+
+import classes from './styles.module.scss'
+
+const DotSelected = () => {
+  return (
+    <button type='button' className={classes.dotSelected} />
```
Comment (nahudalla) [inline src/modules/MainApp/components/Slider/DotSelected.js]: If this button behaves as a kind of "selector" (a slider, checkbox, radio or similar), consider adding the `aria-selected` property to improve usability.

File: src/modules/Store/pages/ProductDetail/Costs/index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      <div className={classes.prices}>
+        <span className={classes.oldPrice}>
+          {currency.ARS(oldPrice)}
+        </span>
+        <span className={classes.currentPrice}>
+          {currency.ARS(price)}
+        </span>
+      </div>
+      <div className={classes.delivery}>
+        <img src={ShippingIcon} alt='shipping-icon' />
```
Comment (nahudalla) [inline src/modules/Store/pages/ProductDetail/Costs/index.js]: `alt` text is human-readable. It should be something like `Shipping icon`. Also, if it's not meaningful content for the user it's best to leave the `alt` prop set as an empty string `alt=''`.

**Review by nahudalla — CHANGES_REQUESTED** [review]

File: src/modules/MainApp/components/Slider/Dot.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import PropTypes from 'prop-types'
+
+const Dot = ({ className, ...props }) => {
+  return (
+    <div type='button' className={className} {...props} />
+  )
+}
+
+Dot.propTypes = {
+  className: PropTypes.oneOfType([PropTypes.string, PropTypes.array])
```
Comment (nahudalla) [inline src/modules/MainApp/components/Slider/Dot.js]: Are you sure React accepts arrays as the `className` prop on an element?

File: src/modules/MainApp/components/Slider/index.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+import Dot from './Dot'
+
+import classes from './styles.module.scss'
+
+const Slider = ({ currentIndex, maxLength }) => {
+  const dotsClasses = []
+  for (let i = 0; i < maxLength; i += 1) {
+    dotsClasses.push(currentIndex === i ? 'dotSelected' : 'dotUnselected')
+  }
```
Comment (nahudalla) [inline src/modules/MainApp/components/Slider/index.js:11]: :thinking: is this really necessary? Why not do it directly in the `map` below? Maybe something like this is better?
```javascript
<Dot
  key={`dot-${index}`}
  className={classnames({
    [classes.dotSelected]: index === currentIndex,
    [classes.dotUnselected]: index !== currentIndex
  })}
/>
```

**Review by nahudalla — APPROVED** [review]

**Pull request merged**